rct_image_tools/src/rct_image_tools/modified_circle_grid_finder.cpp

```diff
@@ -85,65 +86,90 @@ static std::vector<cv::Point2d> extractKeyPoints(const cv::Mat& image, const std
     }
   }
 
-  // OpenCV creates duplicate centers sometimes.  https://github.com/opencv/opencv/issues/4775
-  // Make sure every center has a matching keypoint.  It may be sufficient just to check centers for duplicates.
-  std::vector<cv::Point2d> centers_tmp = centers;
+  // OpenCV creates duplicate feature_coordinates sometimes.  https://github.com/opencv/opencv/issues/4775
+  // Make sure every center has a matching keypoint.  It may be sufficient just to check feature_coordinates for duplicates.
+  std::vector<cv::Point2d> feature_coordinates_tmp = feature_coordinates;
```

Comment:
It seems like `feature_coordinates_tmp` gets created here, but `feature_coordinates` still gets used in the later parts of this function. We should probably change the function signature to take `feature_coordinates` by value so we can modify it in place and then remove `feature_coordinates_tmp`. That way we don't accidentally one of the variables some places and the other variable in other places by mistake

Reply:
The only purpose of `feature_coordinates_tmp` is that for loop of keypoints to check that every feature has a corresponding keypoint, that's why its made as a tmp and then it is empty a few lines later so it can't be used later